### PR TITLE
VIM-4069: VOD Deep Links

### PR DESCRIPTION
**Pull request by ghking**
#### Ticket

https://vimean.atlassian.net/browse/VIM-4069

#### Ticket Summary

This ticket adds convenience method for requesting VOD films and series.

#### Implementation Summary

Added a new method for requesting VOD items.

Associated PR: https://github.vimeows.com/MobileApps/Vimeo-iOS/pull/877

### CODE REVIEW

File: VIMNetworking/Networking/VimeoClient/VIMClient.h [hunk 1]
```diff
@@ -77,12 +77,16 @@
 
 - (nullable id<VIMRequestToken>)activatePictureResourceWithURI:(nonnull NSString *)URI completionBlock:(nonnull VIMRequestCompletionBlock)completionBlock;
 
-#pragma mark - Videos
+#pragma mark - VOD
 
-- (nullable id<VIMRequestToken>)videoWithURI:(nonnull NSString *)URI completionBlock:(nonnull VIMRequestCompletionBlock)completionBlock;
+- (nullable id<VIMRequestToken>)vodItemWithURI:(nonnull NSString *)URI completionBlock:(nonnull VIMRequestCompletionBlock)completionBlock;
```
Comment (alfiehanssen) [inline VIMNetworking/Networking/VimeoClient/VIMClient.h]: @ghking @huebnerob Presumably we will need some form of these in VimeoNetworking. What's the best way to track that work? Create tickets for it now under a VimeoNetworking epic? Something else?

@huebnerob Where did we end up with the de-duplication of the model layer?

Comment (alfiehanssen) [inline VIMNetworking/Networking/VimeoClient/VIMClient.h]: Related to the acronym decision, we have `vodItemWithURI` here, it has two acronyms in it, one is capitalized and the other is not...

File: VIMNetworking/Networking/VimeoClient/VIMClient.m [hunk 2]
```diff
@@ -252,14 +252,14 @@ - (instancetype)initWithBaseURL:(NSURL *)url
...
+#pragma mark - VOD
 
-- (id<VIMRequestToken>)videoWithURI:(NSString *)URI completionBlock:(VIMRequestCompletionBlock)completionBlock
+- (id<VIMRequestToken>)vodItemWithURI:(NSString *)URI completionBlock:(VIMRequestCompletionBlock)completionBlock
 {
     VIMRequestDescriptor *descriptor = [[VIMRequestDescriptor alloc] init];
     descriptor.urlPath = URI;
-    descriptor.modelClass = [VIMVideo class];
-    descriptor.modelKeyPath = @"";
+    descriptor.modelClass = [VIMVODItem class];
```
Comment (alfiehanssen) [inline VIMNetworking/Networking/VimeoClient/VIMClient.m]: Nit: Thoughts on using `VOD` vs. using `onDemand` or something similar. 

`VIMOnDemandVideo`
`VIMVideo`

???

@huebnerob @nicolelehrer @jasonhawkins

**Comment by alfiehanssen** [comment]
DWR Looks fabby gavvy 😁

File: VIMNetworking/Networking/VimeoClient/VIMClient.h (hunk 1 above)
Comment (ghking) [inline VIMNetworking/Networking/VimeoClient/VIMClient.h]: Ill change the VOD to caps as per our convo ayer

**Pull request merged**